applications/CoSimulationApplication/tests/test_rigid_body_solver.py

```diff
@@ -0,0 +1,2047 @@
+                        "process_name": "AssignVectorVariableProcess",
+                        "Parameters": {
+                            "model_part_name": "Main.RootPoint",
+                            "variable_name": "PRESCRIBED_ROTATION",
+                            "interval": [
+                                0,
+                                "End"
+                            ],
+                            "constrained": [
+                                false,
+                                false,
+                                false
+                            ],
+                            "value": [
+                                "0.05*sin(6*t)",
+                                "0.05*sin(8*t)",
+                                "0.05*sin(13*t)"
+                            ]
+                        }
+                    },
+                    {
+                        "python_module": "assign_vector_variable_process",
+                        "kratos_module": "KratosMultiphysics",
+                        "process_name": "AssignVectorVariableProcess",
+                        "Parameters": {
+                            "model_part_name": "Main.RigidBody",
+                            "variable_name": "PRESCRIBED_MOMENT",
+                            "interval": [
+                                0,
+                                "End"
+                            ],
+                            "constrained": [
+                                false,
+                                false,
+                                false
+                            ],
+                            "value": [
+                                "100*sin(4*t)",
+                                "100*sin(3*t)",
+                                "100*sin(10*t)"
+                            ]
+                        }
+                    }
+                ]
+            }
+        }''')
+        Parameters.RecursivelyAddMissingParameters(self.default_parameters)
+        simulation = RigidBodySolver(self.model, Parameters)
+        simulation.Initialize()
+
+        while simulation.main_model_part.ProcessInfo[KM.TIME] < 0.1:
+            simulation.AdvanceInTime(simulation.main_model_part.ProcessInfo[KM.TIME])
+            simulation.InitializeSolutionStep()
+            simulation.Predict()
+            simulation.SolveSolutionStep()
+            simulation.FinalizeSolutionStep()
+            simulation.OutputSolutionStep()
+
+        simulation.AdvanceInTime(simulation.main_model_part.ProcessInfo[KM.TIME])
+        simulation.InitializeSolutionStep()
+        simulation.Predict()
+        simulation.SolveSolutionStep()
+
+        reference = [14.47730397, -954.52993411, 30.97172112, 56.62280302, -20.61982872, 98.25390568]
+        simulation_effective_load = simulation._GetCompleteVector("rigid_body", KMC.EFFECTIVE_FORCE, KMC.EFFECTIVE_MOMENT)
+
+        self.assertVectorAlmostEqual(reference, simulation_effective_load)
```

Comment:
As far as I see, you are running a whole simulation here ryt? I would create the json file seperately and run it as a normal simulation rather than running it with a skeleton like simulation setup.

Reply:
You are right. I already have a simulation test in rbs_test_factory. I will remove this one.